### PR TITLE
chore: change the base image to distroless

### PR DESCRIPTION
**Pull request by yuweizzz**
### Type of change:

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches
- [ ] Documentation
- [ ] Refactor
- [x] Chore
- [ ] CI/CD or Tests

### What this PR does / why we need it:
Related: #1934 

### Pre-submission checklist:

- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Review by lingsamuel — APPROVED** [review]

**Comment by yuweizzz** [comment]
I think this may cause by static image doesn't contains shell, can we try debug image and fix ci failed first ?

**Comment by lingsamuel** [comment]
CI failed again

**Comment by yuweizzz** [comment]
Got it, I will try to fix it.

**Review by lingsamuel — APPROVED** [review]
Thanks!

File: Dockerfile [hunk 1]
```diff
@@ -31,15 +27,11 @@ RUN if [ "$ENABLE_PROXY" = "true" ] ; then go env -w GOPROXY=https://goproxy.cn,
 COPY . .
 RUN --mount=type=cache,target=/root/.cache/go-build make build
 
-FROM centos:centos7
+FROM gcr.io/distroless/static-debian12:debug
```
Comment (Gallardot) [inline Dockerfile:30]: `debug`? Why not use `nonroot`?

Comment (yuweizzz) [reply Dockerfile:30]: you are right, let me adjust it.

**Review by lingsamuel — APPROVED** [review]

**Review by Gallardot — APPROVED** [review]

**Pull request merged**